src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.CoreRT.cs

```diff
@@ -0,0 +1,207 @@
+        {
+            IntPtr ret = IntPtr.Zero;
+
+            int loadWithAlteredPathFlags = 0;
+            bool libNameIsRelativePath = !Path.IsPathFullyQualified(libraryName);
+
+            // P/Invokes are often declared with variations on the actual library name.
+            // For example, it's common to leave off the extension/suffix of the library
+            // even if it has one, or to leave off a prefix like "lib" even if it has one
+            // (both of these are typically done to smooth over cross-platform differences). 
+            // We try to dlopen with such variations on the original.
+            foreach (LibraryNameVariation libraryNameVariation in LibraryNameVariation.DetermineLibraryNameVariations(libraryName, libNameIsRelativePath))
+            {
+                string currLibNameVariation = libraryNameVariation.Prefix + libraryName + libraryNameVariation.Suffix;
+                
+                if (!libNameIsRelativePath)
+                {
+                    int flags = loadWithAlteredPathFlags;
+                    if ((dllImportSearchPathFlags & (int)DllImportSearchPath.UseDllDirectoryForDependencies) != 0)
+                    {
+                        // LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR is the only flag affecting absolute path. Don't OR the flags
+                        // unconditionally as all absolute path P/Invokes could then lose LOAD_WITH_ALTERED_SEARCH_PATH.
+                        flags |= dllImportSearchPathFlags;
+                    }
+
+                    ret = LocalLoadLibraryHelper(currLibNameVariation, flags, ref errorTracker);
+                    if (ret != IntPtr.Zero)
+                    {
+                        return ret;
+                    }
+                }
+                else if ((callingAssembly != null) && searchAssemblyDirectory)
+                {
+                    // Try to load the module alongside the assembly where the PInvoke was declared.
+                    // This only makes sense in dynamic scenarios (JIT/interpreter), so leaving this out for now.
+                }
+
+                ret = LocalLoadLibraryHelper(currLibNameVariation, dllImportSearchPathFlags, ref errorTracker);
+                if (ret != IntPtr.Zero)
+                {
+                    return ret;
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        private static IntPtr LoadFromPath(string libraryName, bool throwOnError)
+        {
+            LoadLibErrorTracker errorTracker = default;
+            IntPtr ret = LocalLoadLibraryHelper(libraryName, 0, ref errorTracker);
+            if (throwOnError && ret == IntPtr.Zero)
+            {
+                errorTracker.Throw(libraryName);
+            }
+
+            return ret;
+        }
+
+        private static IntPtr LocalLoadLibraryHelper(string libraryName, int flags, ref LoadLibErrorTracker errorTracker)
```

Comment:
`LocalLoadLibraryHelper` -> `LoadLibraryHelper` ?

Reply:
This name matches the name of the CoreCLR method. It makes it easier to crossreference the implementation. I can rename it here if you insist though.

Reply:
I would rename it.